compiler/rustc_span/src/hygiene.rs

```diff
@@ -1193,28 +1227,34 @@ pub fn decode_syntax_context<
     Ok(new_ctxt)
 }
 
-fn for_all_ctxts_in<E, F: FnMut((u32, SyntaxContext, &SyntaxContextData)) -> Result<(), E>>(
+fn for_all_ctxts_in<E, F: FnMut(u32, SyntaxContext, SyntaxContextData) -> Result<(), E>>(
```

Comment:
Could you convert this to `impl Trait`?

Reply:
Also, `SyntaxContextData` is relatively heavy, is there any specific reason to pass it by value here?